### PR TITLE
Switch to new promotion authoring

### LINKED ISSUE #202: Promotion Block images & Resizing
[This logic](https://github.com/hlxsites/famous-smoke-cigaradvisor/blob/4dd442210312d5bb5d539c0faaf28bea4b8fa174/cigaradvisor/blocks/promotion-banner/promotion-banner.js#L60-L68) on the promotion banner happens after the image is already loaded to the page. So the space isn't saved correctly  - and the maximum dimensions of the images isn't derived from the original image.

Rather than using a spreadsheet for the promotion source, which has limitations, what if we used a doc with `promotion-entry` blocks structured like:

| Promotion Block | |
| ------- | --- |
| Image |  < img > |
| CTA | < link >  |
| Start | < date > |
| End | < date >|

We can still just link to the word doc and fetch the `.plain.html` to parse and render the random CTA.

@Kronnox - Thoughts?

### PR DESCRIPTION
**Pull request by Kronnox**
Fix #202

Instead of managing the promotion content in the spreadsheet, the speadsheet (promotions.xlsx) is now only used for scheduling. The promotion content is generic (however, in the current use case only the ImageCTA block is used) and authored as fragments (could really be located anywhere but I chose /fragments/promotions/).

Fun fact: This also enables promotions with rotating images on their own, so rotating promotions with rotating images 🎉

Test URLs:
- Before: https://main--famous-smoke-cigaradvisor--hlxsites.hlx.page/cigaradvisor
- After: https://issue-202--famous-smoke-cigaradvisor--hlxsites.hlx.page/cigaradvisor

### CODE REVIEW

**Review by tmaret — DISMISSED** [review]

**Review by bstopp — APPROVED** [review]

**Review by kailasnadh790 — DISMISSED** [review]

**Review by bstopp — CHANGES_REQUESTED** [review]
Something odd is happening - the section is being marked as loaded before this block is finished with its work.

**Comment by kailasnadh790** [comment]
also, observed a height difference
image

**Comment by bstopp** [comment]
yeah, that's related to the same bug i'm seeing. SHould have a fix soon.

**Review by kailasnadh790 — APPROVED** [review]

**Comment by Kronnox** [comment]
> Something odd is happening - the section is being marked as loaded before this block is finished with its work.

@bstopp Ofc you're absolutely right with this one. Somehow forgot, that EDS doesn't like async handling anymore on the blocks. I'll commit some minor changes for readability

File: cigaradvisor/blocks/promotion-banner/promotion-banner.js [hunk 1]
```diff
@@ -57,14 +57,16 @@ async function getActivePromotions(group) {
...
+async function loadPromotionContent(url) {
+  if (!url) {
+    return null;
+  }
+  let path = url;
+  if (!(path.charAt(0) === '/')) {
+    path = new URL(url).pathname;
+  }
+  const fragment = await loadFragment(path);
+  return fragment ? fragment.querySelector('main > .section > div:first-of-type > div:first-of-type') : null;
```
Comment (bstopp) [inline cigaradvisor/blocks/promotion-banner/promotion-banner.js:69]:
```suggestion
  return fragment ? fragment.querySelector('main > .section .block') : null;
```

File: cigaradvisor/blocks/promotion-banner/promotion-banner.js [hunk 2]
```diff
@@ -80,20 +82,7 @@ export default async function decorate(block) {
...
-  // add image and CLS placeholder
-  const picture = createOptimizedPicture(promotion.image, promotion.alt || '');
-  anchor.appendChild(picture);
-  getImageDimensions(picture.querySelector('img')).then((img) => {
-    const ratio = (parseInt(img.height, 10) / parseInt(img.width, 10)) * 100;
-    // noinspection JSUnresolvedVariable
-    picture.style.paddingBottom = `${ratio}%`;
-    anchor.style.maxWidth = `${img.width}px`;
+  await loadPromotionContent(promotion.content).then((content) => {
+    if (content) block.parentElement.replaceWith(content);
```
Comment (bstopp) [inline cigaradvisor/blocks/promotion-banner/promotion-banner.js:86]: I'm somewhat torn on this. This replaces the wrapper with the block - removing the wrapper element from the original dom. I know i put this here, and it is a fix for making sure this CTA is the same height as the other column. but i wonder if there's a css-only way to fix this w/ layout handling.

**Review by bstopp — APPROVED** [review]

**Pull request merged**